RevenueCatUI/Templates/V2/Components/Packages/PurchaseButton/PurchaseButtonComponentView.swift

```diff
@@ -83,6 +76,74 @@ struct PurchaseButtonComponentView: View {
         }
     }
 
+    private func purchase() async throws {
+        guard let action = self.viewModel.action else {
+            try await self.purchaseInApp()
+            return
+        }
+
+        switch action {
+        case .inAppCheckout:
+            try await self.purchaseInApp()
+        case .webCheckout:
+            try await self.purchaseWeb()
+        case .webProductSelection:
+            try await self.purchaseSelectedWebProduct()
+        }
+    }
+
+    private func purchaseInApp() async throws {
+        self.logIfInPreview(package: self.packageContext.package)
+
+        guard !self.purchaseHandler.actionInProgress else { return }
+
+        guard let selectedPackage = self.packageContext.package else {
+            Logger.error(Strings.no_selected_package_found)
+            return
+        }
+
+        _ = try await self.purchaseHandler.purchase(package: selectedPackage)
+    }
+
+    private func purchaseWeb() async throws {
+        self.logIfInPreview(package: self.packageContext.package)
+
+        guard let webCheckoutUrl = self.packageContext.package?.webCheckoutUrl else {
+            Logger.error(Strings.no_web_checkout_url_found)
+            return
+        }
+
+        self.openWebPaywallLink(url: webCheckoutUrl, method: .externalBrowser)
+    }
+
+    private func purchaseSelectedWebProduct() async throws {
+        self.logIfInPreview(package: self.packageContext.package)
+
+        guard let webCheckoutUrl = self.viewModel.offeringWebCheckoutUrl else {
```

Comment:
Hmm so this will initiate the purchase with the web package selector? Should this method name be `purchaseWeb` and the `purchaseWeb` method be `purchaseSelectedWebProduct`? (the other way around). That makes more sense to me...